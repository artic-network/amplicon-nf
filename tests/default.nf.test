nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"

    test("-profile test") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/amplicon-nf_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    "artic-sars-cov-2-400-v3.0.0_amplicon-nf_run-report.html - minimum file size: ${file("$outputDir/artic-sars-cov-2_400_v3.0.0_amplicon-nf_run-report.html").size() >= 4000000}",
                    "test_custom_scheme_name_amplicon-nf_run-report.html - minimum file size: ${file("$outputDir/test_custom_scheme_name_amplicon-nf_run-report.html").size() >= 4000000}",
                    file("$outputDir/MN908947.3_artic-sars-cov-2_400_v3.0.0_aligned-consensus.fasta").exists(),
                    file("$outputDir/MN908947.3_test_custom_scheme_name_aligned-consensus.fasta").exists(),
                    file("$outputDir/artic-sars-cov-2_400_v3.0.0_qc_results.tsv").exists(),
                    file("$outputDir/test_custom_scheme_name_qc_results.tsv").exists(),
                    "illumina_amplicon_test/illumina_amplicon_test_amplicon-nf_sample-report.html - minimum file size: ${file("$outputDir/illumina_amplicon_test/illumina_amplicon_test_amplicon-nf_sample-report.html").size() >= 4000000}",
                    file("$outputDir/illumina_amplicon_test/illumina_amplicon_test.amplicon_depths.tsv").exists(),
                    file("$outputDir/illumina_amplicon_test/illumina_amplicon_test.consensus.fasta").exists(),
                    file("$outputDir/illumina_amplicon_test/illumina_amplicon_test.primertrimmed.sorted.bam").exists(),
                    file("$outputDir/illumina_amplicon_test/illumina_amplicon_test.vcf.gz").exists(),
                    "nanopore_amplicon_test/nanopore_amplicon_test_amplicon-nf-report.html - minimum file size: ${file("$outputDir/nanopore_amplicon_test/nanopore_amplicon_test_amplicon-nf-report.html").size() >= 4000000}",
                    file("$outputDir/nanopore_amplicon_test/nanopore_amplicon_test.amplicon_depths.tsv").exists(),
                    file("$outputDir/nanopore_amplicon_test/nanopore_amplicon_test.consensus.fasta").exists(),
                    file("$outputDir/nanopore_amplicon_test/nanopore_amplicon_test.vcf.gz").exists(),
                    file("$outputDir/nanopore_amplicon_test/nanopore_amplicon_test.primertrimmed.sorted.bam").exists(),
                    file("$outputDir/nanopore_amplicon_test/nanopore_amplicon_test.sorted.bam").exists(),
                    file("$outputDir/illumina_fuzzy_match/illumina_fuzzy_match.consensus.fasta").exists(),
                    file("$outputDir/nanopore_fuzzy_match/nanopore_fuzzy_match.consensus.fasta").exists(),
                    ).match()
                }
            )
        }
    }
}
