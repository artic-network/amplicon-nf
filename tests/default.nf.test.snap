{
    "-profile test": {
        "content": [
            98,
            {
                "ARTIC_ALIGN_TRIM": {
                    "artic": "1.7.4"
                },
                "ARTIC_GET_MODELS": {
                    "artic": "1.7.4"
                },
                "ARTIC_GET_SCHEME": {
                    "artic": "1.7.4"
                },
                "ARTIC_GUPPYPLEX": {
                    "artic": "1.7.4"
                },
                "ARTIC_MINION": {
                    "artic": "1.7.4"
                },
                "BCFTOOLS_CONSENSUS_AMBIGUOUS": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_CONSENSUS_FIXED": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_NORM": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_VIEW": {
                    "bcftools": 1.21
                },
                "BWAMEM2_INDEX": {
                    "bwamem2": "2.2.1"
                },
                "BWAMEM2_MEM": {
                    "bwamem2": "2.2.1",
                    "samtools": 1.21
                },
                "CAT_CAT": {
                    "pigz": "2.3.4"
                },
                "FREEBAYES": {
                    "freebayes": "1.3.6"
                },
                "GENERATE_RUN_REPORT": {
                    "plotly": "6.1.2",
                    "primalbedtools": 0.9,
                    "pandas": "2.3.0",
                    "jinja2": "3.1.6"
                },
                "GENERATE_SAMPLE_REPORT": {
                    "plotly": "6.1.2",
                    "primalbedtools": 0.9,
                    "pandas": "2.3.0",
                    "jinja2": "3.1.6"
                },
                "MAFFT_ALIGN": {
                    "mafft": 7.52,
                    "pigz": "2.8)"
                },
                "PROCESS_GVCF": {
                    "pysam": "0.23.0"
                },
                "SAMTOOLS_COVERAGE": {
                    "samtools": 1.21
                },
                "SAMTOOLS_DEPTH": {
                    "samtools": 1.21
                },
                "SAMTOOLS_FAIDX": {
                    "samtools": 1.21
                },
                "SAMTOOLS_INDEX": {
                    "samtools": 1.21
                },
                "SEQKIT_GREP_FASTAS": {
                    "seqkit": "2.9.0"
                },
                "SEQKIT_GREP_REFS": {
                    "seqkit": "2.9.0"
                },
                "SEQKIT_REPLACE_ILLUMINA": {
                    "seqkit": "2.9.0"
                },
                "SEQKIT_REPLACE_ONT": {
                    "seqkit": "2.9.0"
                },
                "TRIMMOMATIC": {
                    "trimmomatic": 0.39
                },
                "Workflow": {
                    "artic-network/amplicon-nf": "v0.2.1dev"
                }
            },
            [
                "MN908947.3_artic-sars-cov-2_400_v3.0.0_aligned-consensus.fasta",
                "MN908947.3_test_custom_scheme_name_aligned-consensus.fasta",
                "artic-sars-cov-2_400_v3.0.0_amplicon-nf_run-report.html",
                "artic-sars-cov-2_400_v3.0.0_qc_results.tsv",
                "illumina_amplicon_test",
                "illumina_amplicon_test/illumina_amplicon_test.align_trim_report.csv",
                "illumina_amplicon_test/illumina_amplicon_test.amplicon_depths.tsv",
                "illumina_amplicon_test/illumina_amplicon_test.consensus.fasta",
                "illumina_amplicon_test/illumina_amplicon_test.primertrimmed.sorted.bam",
                "illumina_amplicon_test/illumina_amplicon_test.sorted.bam",
                "illumina_amplicon_test/illumina_amplicon_test.vcf.gz",
                "illumina_amplicon_test/illumina_amplicon_test.vcf.gz.tbi",
                "illumina_amplicon_test/illumina_amplicon_test_amplicon-nf_sample-report.html",
                "illumina_custom_scheme_Test",
                "illumina_custom_scheme_Test/illumina_custom_scheme_Test.align_trim_report.csv",
                "illumina_custom_scheme_Test/illumina_custom_scheme_Test.amplicon_depths.tsv",
                "illumina_custom_scheme_Test/illumina_custom_scheme_Test.consensus.fasta",
                "illumina_custom_scheme_Test/illumina_custom_scheme_Test.primertrimmed.sorted.bam",
                "illumina_custom_scheme_Test/illumina_custom_scheme_Test.sorted.bam",
                "illumina_custom_scheme_Test/illumina_custom_scheme_Test.vcf.gz",
                "illumina_custom_scheme_Test/illumina_custom_scheme_Test.vcf.gz.tbi",
                "illumina_custom_scheme_Test/illumina_custom_scheme_Test_amplicon-nf_sample-report.html",
                "illumina_fuzzy_match",
                "illumina_fuzzy_match/illumina_fuzzy_match.align_trim_report.csv",
                "illumina_fuzzy_match/illumina_fuzzy_match.amplicon_depths.tsv",
                "illumina_fuzzy_match/illumina_fuzzy_match.consensus.fasta",
                "illumina_fuzzy_match/illumina_fuzzy_match.primertrimmed.sorted.bam",
                "illumina_fuzzy_match/illumina_fuzzy_match.sorted.bam",
                "illumina_fuzzy_match/illumina_fuzzy_match.vcf.gz",
                "illumina_fuzzy_match/illumina_fuzzy_match.vcf.gz.tbi",
                "illumina_fuzzy_match/illumina_fuzzy_match_amplicon-nf_sample-report.html",
                "nanopore_amplicon_test",
                "nanopore_amplicon_test/nanopore_amplicon_test.amplicon_depths.tsv",
                "nanopore_amplicon_test/nanopore_amplicon_test.consensus.fasta",
                "nanopore_amplicon_test/nanopore_amplicon_test.fastq.gz",
                "nanopore_amplicon_test/nanopore_amplicon_test.minion.log.txt",
                "nanopore_amplicon_test/nanopore_amplicon_test.primertrimmed.sorted.bam",
                "nanopore_amplicon_test/nanopore_amplicon_test.primertrimmed.sorted.bam.bai",
                "nanopore_amplicon_test/nanopore_amplicon_test.sorted.bam",
                "nanopore_amplicon_test/nanopore_amplicon_test.sorted.bam.bai",
                "nanopore_amplicon_test/nanopore_amplicon_test.trimmed.rg.sorted.bam",
                "nanopore_amplicon_test/nanopore_amplicon_test.trimmed.rg.sorted.bam.bai",
                "nanopore_amplicon_test/nanopore_amplicon_test.vcf.gz",
                "nanopore_amplicon_test/nanopore_amplicon_test_amplicon-nf_sample-report.html",
                "nanopore_amplicon_test/primer.bed",
                "nanopore_amplicon_test/reference.fasta",
                "nanopore_custom_scheme_test",
                "nanopore_custom_scheme_test/nanopore_custom_scheme_test.amplicon_depths.tsv",
                "nanopore_custom_scheme_test/nanopore_custom_scheme_test.consensus.fasta",
                "nanopore_custom_scheme_test/nanopore_custom_scheme_test.fastq.gz",
                "nanopore_custom_scheme_test/nanopore_custom_scheme_test.minion.log.txt",
                "nanopore_custom_scheme_test/nanopore_custom_scheme_test.primertrimmed.sorted.bam",
                "nanopore_custom_scheme_test/nanopore_custom_scheme_test.primertrimmed.sorted.bam.bai",
                "nanopore_custom_scheme_test/nanopore_custom_scheme_test.sorted.bam",
                "nanopore_custom_scheme_test/nanopore_custom_scheme_test.sorted.bam.bai",
                "nanopore_custom_scheme_test/nanopore_custom_scheme_test.trimmed.rg.sorted.bam",
                "nanopore_custom_scheme_test/nanopore_custom_scheme_test.trimmed.rg.sorted.bam.bai",
                "nanopore_custom_scheme_test/nanopore_custom_scheme_test.vcf.gz",
                "nanopore_custom_scheme_test/nanopore_custom_scheme_test_amplicon-nf_sample-report.html",
                "nanopore_custom_scheme_test/primer.bed",
                "nanopore_custom_scheme_test/reference.fasta",
                "nanopore_empty_files",
                "nanopore_empty_files/nanopore_empty_files.amplicon_depths.tsv",
                "nanopore_empty_files/nanopore_empty_files.consensus.fasta",
                "nanopore_empty_files/nanopore_empty_files.fastq.gz",
                "nanopore_empty_files/nanopore_empty_files.minion.log.txt",
                "nanopore_empty_files/nanopore_empty_files.primertrimmed.sorted.bam",
                "nanopore_empty_files/nanopore_empty_files.primertrimmed.sorted.bam.bai",
                "nanopore_empty_files/nanopore_empty_files.sorted.bam",
                "nanopore_empty_files/nanopore_empty_files.sorted.bam.bai",
                "nanopore_empty_files/nanopore_empty_files.trimmed.rg.sorted.bam",
                "nanopore_empty_files/nanopore_empty_files.trimmed.rg.sorted.bam.bai",
                "nanopore_empty_files/nanopore_empty_files.vcf.gz",
                "nanopore_empty_files/nanopore_empty_files_amplicon-nf_sample-report.html",
                "nanopore_empty_files/primer.bed",
                "nanopore_empty_files/reference.fasta",
                "nanopore_fuzzy_match",
                "nanopore_fuzzy_match/nanopore_fuzzy_match.amplicon_depths.tsv",
                "nanopore_fuzzy_match/nanopore_fuzzy_match.consensus.fasta",
                "nanopore_fuzzy_match/nanopore_fuzzy_match.fastq.gz",
                "nanopore_fuzzy_match/nanopore_fuzzy_match.minion.log.txt",
                "nanopore_fuzzy_match/nanopore_fuzzy_match.primertrimmed.sorted.bam",
                "nanopore_fuzzy_match/nanopore_fuzzy_match.primertrimmed.sorted.bam.bai",
                "nanopore_fuzzy_match/nanopore_fuzzy_match.sorted.bam",
                "nanopore_fuzzy_match/nanopore_fuzzy_match.sorted.bam.bai",
                "nanopore_fuzzy_match/nanopore_fuzzy_match.trimmed.rg.sorted.bam",
                "nanopore_fuzzy_match/nanopore_fuzzy_match.trimmed.rg.sorted.bam.bai",
                "nanopore_fuzzy_match/nanopore_fuzzy_match.vcf.gz",
                "nanopore_fuzzy_match/nanopore_fuzzy_match_amplicon-nf_sample-report.html",
                "nanopore_fuzzy_match/primer.bed",
                "nanopore_fuzzy_match/reference.fasta",
                "pipeline_info",
                "pipeline_info/amplicon-nf_software_mqc_versions.yml",
                "test_custom_scheme_name_amplicon-nf_run-report.html",
                "test_custom_scheme_name_qc_results.tsv"
            ],
            "artic-sars-cov-2-400-v3.0.0_amplicon-nf_run-report.html - minimum file size: true",
            "test_custom_scheme_name_amplicon-nf_run-report.html - minimum file size: true",
            true,
            true,
            true,
            true,
            "illumina_amplicon_test/illumina_amplicon_test_amplicon-nf_sample-report.html - minimum file size: true",
            true,
            true,
            true,
            true,
            "nanopore_amplicon_test/nanopore_amplicon_test_amplicon-nf-report.html - minimum file size: false",
            true,
            true,
            true,
            true,
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-08-20T14:25:04.899043"
    }
}